update_blocklist.nix

```
@@ -40,16 +40,43 @@ in
   # Create an ip set and add each ip to it one by one
 
   # IPv4 and IPv6 regex patterns with CIDR notation support - WARNING: might not be correct for all IPs (e.g. ignore valid ones or accept wrong ones), but seems to work fine
+  # Also supports hosts files, as well as plain lists of domains.
   ipv4_regex="^([0-9]{1,3}\.){3}[0-9]{1,3}(\/[0-9]{1,2})?$"
   ipv6_regex="^([0-9a-fA-F:]+::?[0-9a-fA-F]*)+(\/[0-9]{1,3})?$"
+  domain_regex='^[a-zA-Z0-9.-]*\.[a-zA-Z][a-zA-Z0-9.-]+$'
+  host_regex="^0\\.0\\.0\\.0 ''${domain_regex:1}"
+
+  blockIPv4 () {
+      echo -exist add "${ipSetName}" "$1"
+  }
+
+  blockIPv6 () {
+      echo -exist add "${ipV6SetName}" "$1"
+  }
+
+  blockDomain () {
+      ${dig}/bin/dig "$1" A +short |
+      while IFS= read -r IP; do
+          blockIPv4 "$IP"
+      done
+
+      ${dig}/bin/dig $1 AAAA +short |
+      while IFS= read -r IP; do
+          blockIPv6 "$IP"
```

Now we only need to make sure that we filter out the non-IP addresses:
```sh
> dig mclean.f.360.cn A +short                     
i18nv2.mclean.50union.com.
104.192.110.73
8.222.197.237
```
```suggestion
          if [[ $IP =~ $ipv4_regex ]]; then
              blockIPv4 "$IP"
          fi
      done

      ${dig}/bin/dig $1 AAAA +short |
      while IFS= read -r IP; do
          if [[ "$IP" =~ $ipv6_regex ]]; then
              blockIPv6 "$IP"
          fi
```